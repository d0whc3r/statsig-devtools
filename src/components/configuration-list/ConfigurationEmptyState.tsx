import React from 'react'

interface ConfigurationEmptyStateProps {
  hasActiveFilters: boolean
}

/**
 * Empty state component for configuration list
 */
export function ConfigurationEmptyState({ hasActiveFilters }: ConfigurationEmptyStateProps) {
  return (
    <div className="p-8 text-center">
      <div className="text-gray-400">
        <svg className="mx-auto h-12 w-12" fill="none" viewBox="0 0 24 24" stroke="currentColor">
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            d="M9 5H7a2 2 0 00-2 2v10a2 2 0 002 2h8a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2"
          />
        </svg>
      </div>
      <h3 className="mt-2 text-sm font-medium text-gray-900">No configurations found</h3>
      <p className="mt-1 text-sm text-gray-500">
        {hasActiveFilters ? 'Try adjusting your search or filters.' : 'No configurations available.'}
      </p>
    </div>
  )
}
