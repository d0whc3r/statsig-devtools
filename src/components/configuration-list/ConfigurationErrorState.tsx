import React from 'react'

interface ConfigurationErrorStateProps {
  error: string
}

/**
 * Error state component for configuration list
 */
export function ConfigurationErrorState({ error }: ConfigurationErrorStateProps) {
  return (
    <div className="p-4">
      <div className="rounded-md bg-red-50 p-4">
        <div className="flex">
          <div className="ml-3">
            <h3 className="text-sm font-medium text-red-800">Error loading configurations</h3>
            <div className="mt-2 text-sm text-red-700">
              <p>{error}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}
