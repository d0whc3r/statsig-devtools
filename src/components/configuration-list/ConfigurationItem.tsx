import React, { useCallback } from 'react'

import { formatTypeName, formatValue, getTypeBadgeClass } from '../../utils/configuration-formatters'
import { ConfigurationStatusIndicator } from '../ConfigurationStatusIndicator'

import type { EvaluationResult } from '../../services/statsig-integration'
import type { StatsigConfigurationItem } from '../../types'

interface ConfigurationItemProps {
  config: StatsigConfigurationItem
  isSelected: boolean
  result?: EvaluationResult
  onSelect: (config: StatsigConfigurationItem) => void
  hasOverrides: boolean
  overrideCount: number
}

/**
 * Individual configuration item component
 */
export const ConfigurationItem = React.memo(
  ({ config, isSelected, result, onSelect, hasOverrides, overrideCount }: ConfigurationItemProps) => {
    const handleClick = useCallback(() => {
      onSelect(config)
    }, [config, onSelect])

    const handleKeyDown = useCallback(
      (event: React.KeyboardEvent) => {
        if (event.key === 'Enter' || event.key === ' ') {
          event.preventDefault()
          handleClick()
        }
      },
      [handleClick],
    )

    return (
      <div
        onClick={handleClick}
        onKeyDown={handleKeyDown}
        role="button"
        tabIndex={0}
        className={`card-professional mx-2 my-1.5 cursor-pointer p-3 ${isSelected ? 'selected' : ''}`}
      >
        <div className="flex items-start justify-between">
          <div className="min-w-0 flex-1">
            <div className="mb-2 flex items-center gap-2">
              <ConfigurationStatusIndicator result={result} size="sm" />
              <h3 className="truncate text-sm font-semibold text-gray-900">{config.name}</h3>
            </div>

            <div className="mb-2 flex flex-wrap items-center gap-1">
              <span className={`${getTypeBadgeClass(config.type)} text-xs`}>{formatTypeName(config.type)}</span>

              {!config.enabled && (
                <span className="type-badge bg-gray-100 text-xs text-gray-800">
                  <svg className="h-2.5 w-2.5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={2}
                      d="M18.364 18.364A9 9 0 005.636 5.636m12.728 12.728L5.636 5.636m12.728 12.728L18.364 5.636M5.636 18.364l12.728-12.728"
                    />
                  </svg>
                  Disabled
                </span>
              )}

              {hasOverrides && (
                <span className="type-badge border border-orange-200 bg-orange-100 text-xs text-orange-800">
                  <svg className="h-2.5 w-2.5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={2}
                      d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.964-.833-2.732 0L3.732 16.5c-.77.833.192 2.5 1.732 2.5z"
                    />
                  </svg>
                  {overrideCount}
                </span>
              )}
            </div>

            {result && (
              <div className="rounded-md bg-gray-50 px-2 py-1.5 text-xs text-gray-600">
                {config.type === 'feature_gate' && (
                  <span className="font-medium">
                    <span className={result.value ? 'text-green-700' : 'text-red-700'}>
                      {formatValue(result.value)}
                    </span>
                  </span>
                )}
                {config.type === 'experiment' && result.groupName && (
                  <span className="font-medium">
                    Group: <span className="text-purple-700">{result.groupName}</span>
                  </span>
                )}
                {config.type === 'dynamic_config' && <span className="font-medium text-green-700">Ready</span>}
              </div>
            )}
          </div>

          <div className="ml-2 flex-shrink-0">
            <svg className="h-4 w-4 text-gray-400" fill="none" viewBox="0 0 24 24" stroke="currentColor">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5l7 7-7 7" />
            </svg>
          </div>
        </div>
      </div>
    )
  },
)

ConfigurationItem.displayName = 'ConfigurationItem'
