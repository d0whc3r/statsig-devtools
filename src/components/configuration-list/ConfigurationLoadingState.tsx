import React from 'react'

import { LoadingSpinner } from '../LoadingSpinner'

/**
 * Loading state component for configuration list
 */
export function ConfigurationLoadingState() {
  return (
    <div className="flex items-center justify-center p-8">
      <div className="text-center">
        <LoadingSpinner size="md" />
        <p className="mt-2 text-sm text-gray-600">Loading configurations...</p>
      </div>
    </div>
  )
}
