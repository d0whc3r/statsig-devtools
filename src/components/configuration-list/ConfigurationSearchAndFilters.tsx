import React from 'react'

import type { FilterStatus, FilterType } from './useConfigurationFilters'

interface ConfigurationSearchAndFiltersProps {
  searchQuery: string
  filterType: FilterType
  filterStatus: FilterStatus
  onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  onFilterTypeChange: (filterType: FilterType) => void
  onFilterStatusChange: (filterStatus: FilterStatus) => void
  onClearFilters: () => void
  hasActiveFilters: boolean
  totalConfigurations: number
  filteredCount: number
}

/**
 * Search and filter controls component
 */
export function ConfigurationSearchAndFilters({
  searchQuery,
  filterType,
  filterStatus,
  onSearchChange,
  onFilterTypeChange,
  onFilterStatusChange,
  onClearFilters,
  hasActiveFilters,
  totalConfigurations,
  filteredCount,
}: ConfigurationSearchAndFiltersProps) {
  return (
    <div className="flex-shrink-0 border-b border-gray-200 p-3">
      <div className="mb-3">
        <div className="relative">
          <div className="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3">
            <svg className="h-4 w-4 text-slate-400" fill="none" viewBox="0 0 24 24" stroke="currentColor">
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
              />
            </svg>
          </div>
          <input
            type="text"
            placeholder="Search..."
            value={searchQuery}
            onChange={onSearchChange}
            className="input-professional w-full py-2.5 pr-4 pl-10 text-sm"
          />
        </div>
      </div>

      <div className="flex items-center gap-2">
        <select
          value={filterType}
          onChange={(e) => onFilterTypeChange(e.target.value as FilterType)}
          className="input-professional min-w-0 flex-1 px-3 py-1.5 text-xs font-medium"
        >
          <option value="all">All Types</option>
          <option value="feature_gate">üö™ Gates</option>
          <option value="experiment">üß™ Experiments</option>
          <option value="dynamic_config">‚öôÔ∏è Configs</option>
        </select>

        <select
          value={filterStatus}
          onChange={(e) => onFilterStatusChange(e.target.value as FilterStatus)}
          className="input-professional min-w-0 flex-1 px-3 py-1.5 text-xs font-medium"
        >
          <option value="all">All Status</option>
          <option value="passed">‚úÖ Passed</option>
          <option value="failed">‚ùå Failed</option>
        </select>

        {hasActiveFilters && (
          <button
            onClick={onClearFilters}
            className="icon-button-professional h-8 w-8 flex-shrink-0 text-slate-600 hover:text-red-600"
            title="Clear filters"
          >
            <svg className="h-3.5 w-3.5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M6 18L18 6M6 6l12 12" />
            </svg>
          </button>
        )}
      </div>

      <div className="mt-3 flex items-center justify-between">
        <div className="flex items-center gap-2 text-xs text-slate-600">
          <svg className="h-3.5 w-3.5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
              d="M9 5H7a2 2 0 00-2 2v10a2 2 0 002 2h8a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2"
            />
          </svg>
          <span className="font-medium">
            {filteredCount === totalConfigurations ? (
              <span>{totalConfigurations} configurations</span>
            ) : (
              <span>
                <span className="font-semibold text-blue-600">{filteredCount}</span> of {totalConfigurations}
              </span>
            )}
          </span>
        </div>
        {hasActiveFilters && <div className="text-xs font-medium text-orange-600">Filtered</div>}
      </div>
    </div>
  )
}
