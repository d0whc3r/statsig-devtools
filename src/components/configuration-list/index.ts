export { ConfigurationEmptyState } from './ConfigurationEmptyState'
export { ConfigurationErrorState } from './ConfigurationErrorState'
export { ConfigurationItem } from './ConfigurationItem'
export { ConfigurationLoadingState } from './ConfigurationLoadingState'
export { ConfigurationSearchAndFilters } from './ConfigurationSearchAndFilters'
export type { FilterStatus, FilterType } from './useConfigurationFilters'
export { useConfigurationFilters } from './useConfigurationFilters'
export { useConfigurationOverrides } from './useConfigurationOverrides'
