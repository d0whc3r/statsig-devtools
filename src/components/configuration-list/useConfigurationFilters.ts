import { useCallback, useMemo, useState } from 'react'

import { useDebouncedSearch } from '../VirtualizedList'

import type { EvaluationResult } from '../../services/unified-statsig-api'
import type { StatsigConfigurationItem } from '../../types'
import type React from 'react'

export type FilterType = 'all' | 'feature_gate' | 'experiment' | 'dynamic_config'
export type FilterStatus = 'all' | 'passed' | 'failed'

/**
 * Hook to manage configuration filtering and searching
 */
export const useConfigurationFilters = (
  configurations: StatsigConfigurationItem[],
  evaluationResults: Map<string, EvaluationResult>,
) => {
  const [debouncedSearchQuery, setSearchQuery, immediateSearchQuery] = useDebouncedSearch('', 300)
  const [filterType, setFilterType] = useState<FilterType>('all')
  const [filterStatus, setFilterStatus] = useState<FilterStatus>('all')

  /**
   * Filter configurations by type
   */
  const filterByType = useCallback(
    (configs: StatsigConfigurationItem[]) =>
      filterType === 'all' ? configs : configs.filter((config) => config.type === filterType),
    [filterType],
  )

  /**
   * Filter configurations by evaluation status
   */
  const filterByStatus = useCallback(
    (configs: StatsigConfigurationItem[]) => {
      if (filterStatus === 'all') return configs

      return configs.filter((config) => {
        const result = evaluationResults.get(config.name)
        if (!result) return filterStatus === 'failed'
        return filterStatus === 'passed' ? result.passed : !result.passed
      })
    },
    [filterStatus, evaluationResults],
  )

  /**
   * Filter configurations by search query (searches both name and ID)
   */
  const filterBySearch = useCallback(
    (configs: StatsigConfigurationItem[]) => {
      if (!debouncedSearchQuery.trim()) return configs

      const query = debouncedSearchQuery.toLowerCase().trim()
      return configs.filter((config) => {
        const nameMatch = config.name.toLowerCase().includes(query)
        const idMatch = config.id ? config.id.toLowerCase().includes(query) : false
        return nameMatch || idMatch
      })
    },
    [debouncedSearchQuery],
  )

  /**
   * Filter and search configurations
   */
  const filteredConfigurations = useMemo(() => {
    let filtered = configurations
    filtered = filterByType(filtered)
    filtered = filterByStatus(filtered)
    filtered = filterBySearch(filtered)
    return filtered
  }, [configurations, filterByType, filterByStatus, filterBySearch])

  /**
   * Handle search input change with debouncing
   */
  const handleSearchChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      setSearchQuery(event.target.value)
    },
    [setSearchQuery],
  )

  /**
   * Clear all filters
   */
  const clearFilters = useCallback(() => {
    setSearchQuery('')
    setFilterType('all')
    setFilterStatus('all')
  }, [setSearchQuery])

  /**
   * Check if any filters are active
   */
  const hasActiveFilters = useMemo(
    () => immediateSearchQuery.trim() !== '' || filterType !== 'all' || filterStatus !== 'all',
    [immediateSearchQuery, filterType, filterStatus],
  )

  return {
    searchQuery: immediateSearchQuery,
    filterType,
    filterStatus,
    filteredConfigurations,
    handleSearchChange,
    setFilterType,
    setFilterStatus,
    clearFilters,
    hasActiveFilters,
  }
}
