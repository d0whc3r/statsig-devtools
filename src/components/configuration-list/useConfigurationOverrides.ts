import { useCallback } from 'react'

import type { StorageOverride } from '../../types'

/**
 * Hook to manage configuration overrides logic
 */
export const useConfigurationOverrides = (activeOverrides: StorageOverride[]) => {
  /**
   * Check if a configuration has active overrides
   */
  const hasOverrides = useCallback(
    (configName: string) =>
      activeOverrides.some((override) => {
        // Check for Statsig-specific overrides first
        const extendedOverride = override as StorageOverride & { featureName?: string }
        if (extendedOverride.featureName) {
          return extendedOverride.featureName === configName
        }
        // Fallback to legacy key/value matching
        return override.key.includes(configName) || override.value?.toString().includes(configName)
      }),
    [activeOverrides],
  )

  /**
   * Get override count for a configuration
   */
  const getOverrideCount = useCallback(
    (configName: string) =>
      activeOverrides.filter((override) => {
        // Check for Statsig-specific overrides first
        const extendedOverride = override as StorageOverride & { featureName?: string }
        if (extendedOverride.featureName) {
          return extendedOverride.featureName === configName
        }
        // Fallback to legacy key/value matching
        return override.key.includes(configName) || override.value?.toString().includes(configName)
      }).length,
    [activeOverrides],
  )

  return {
    hasOverrides,
    getOverrideCount,
  }
}
