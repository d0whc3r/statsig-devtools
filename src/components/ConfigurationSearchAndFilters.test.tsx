import { describe, expect, it, vi } from 'vitest'

import { ConfigurationSearchAndFilters } from './configuration-list/ConfigurationSearchAndFilters'

import { fireEvent, render, screen } from '@testing-library/react'

describe('ConfigurationSearchAndFilters', () => {
  const defaultProps = {
    searchQuery: '',
    onSearchChange: vi.fn(),
    filterType: 'all' as const,
    onFilterTypeChange: vi.fn(),
    filterStatus: 'all' as const,
    onFilterStatusChange: vi.fn(),
    hasActiveFilters: false,
    onClearFilters: vi.fn(),
    totalConfigurations: 10,
    filteredCount: 10,
  }

  it('should match snapshot with default props', () => {
    const { container } = render(<ConfigurationSearchAndFilters {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('should render search input with correct placeholder', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} />)

    const searchInput = screen.getByPlaceholderText('Search...')
    expect(searchInput).toBeInTheDocument()
  })

  it('should render filter dropdowns', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} />)

    const selects = screen.getAllByRole('combobox')
    expect(selects).toHaveLength(2)
  })

  it('should call onSearchChange when typing in search input', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} />)

    const searchInput = screen.getByPlaceholderText('Search...')
    fireEvent.change(searchInput, { target: { value: 'test search' } })

    expect(defaultProps.onSearchChange).toHaveBeenCalled()
  })

  it('should display search value in input', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} searchQuery="test value" />)

    const searchInput = screen.getByDisplayValue('test value')
    expect(searchInput).toBeInTheDocument()
  })

  it('should display configuration count', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} totalConfigurations={25} filteredCount={15} />)

    // Text may appear in multiple nested elements; allow multiple matches
    const matches = screen.getAllByText((_, node) => (node?.textContent || '').trim() === '15 of 25')
    expect(matches.length).toBeGreaterThan(0)
  })

  it('should handle filter type changes', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} />)

    const selects = screen.getAllByRole('combobox')
    const typeSelect = selects[0] // First select should be type filter

    fireEvent.change(typeSelect, { target: { value: 'feature_gate' } })

    expect(defaultProps.onFilterTypeChange).toHaveBeenCalledWith('feature_gate')
  })

  it('should handle filter status changes', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} />)

    const selects = screen.getAllByRole('combobox')
    const statusSelect = selects[1] // Second select should be status filter

    fireEvent.change(statusSelect, { target: { value: 'passed' } })

    expect(defaultProps.onFilterStatusChange).toHaveBeenCalledWith('passed')
  })

  it('should render clear button when hasActiveFilters is true', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} hasActiveFilters />)

    const clearButton = screen.getByRole('button', { name: /clear/i })
    expect(clearButton).toBeInTheDocument()
  })

  it('should not render clear button when hasActiveFilters is false', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} />)

    const clearButton = screen.queryByRole('button', { name: /clear/i })
    expect(clearButton).not.toBeInTheDocument()
  })

  it('should call onClearFilters when clicking clear button', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} hasActiveFilters />)

    const clearButton = screen.getByRole('button', { name: /clear/i })
    fireEvent.click(clearButton)

    expect(defaultProps.onClearFilters).toHaveBeenCalled()
  })

  it('should display correct filter type value', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} filterType="feature_gate" />)

    const typeSelect = screen.getByDisplayValue(/Gates$/)
    expect(typeSelect).toBeInTheDocument()
  })

  it('should display correct filter status value', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} filterStatus="passed" />)

    const statusSelect = screen.getByDisplayValue(/Passed$/)
    expect(statusSelect).toBeInTheDocument()
  })

  // Additional tests for better branch coverage
  it('should handle all filter type options', () => {
    const filterTypes = ['all', 'feature_gate', 'dynamic_config', 'experiment'] as const

    filterTypes.forEach((filterType) => {
      const { unmount } = render(<ConfigurationSearchAndFilters {...defaultProps} filterType={filterType} />)

      // Use a more specific selector to avoid conflicts
      const selects = screen.getAllByRole('combobox')
      const typeSelect = selects[0] // First select is always the type filter

      if (filterType === 'all') {
        expect(typeSelect).toHaveDisplayValue('All Types')
      } else if (filterType === 'feature_gate') {
        expect(typeSelect).toHaveDisplayValue('🚪 Gates')
      } else if (filterType === 'dynamic_config') {
        expect(typeSelect).toHaveDisplayValue('⚙️ Configs')
      } else if (filterType === 'experiment') {
        expect(typeSelect).toHaveDisplayValue('🧪 Experiments')
      }

      unmount()
    })
  })

  it('should handle all filter status options', () => {
    const filterStatuses = ['all', 'passed', 'failed'] as const

    filterStatuses.forEach((filterStatus) => {
      const { unmount } = render(<ConfigurationSearchAndFilters {...defaultProps} filterStatus={filterStatus} />)

      // Use a more specific selector to avoid conflicts
      const selects = screen.getAllByRole('combobox')
      const statusSelect = selects[1] // Second select is always the status filter

      if (filterStatus === 'all') {
        expect(statusSelect).toHaveDisplayValue('All Status')
      } else if (filterStatus === 'passed') {
        expect(statusSelect).toHaveDisplayValue('✅ Passed')
      } else if (filterStatus === 'failed') {
        expect(statusSelect).toHaveDisplayValue('❌ Failed')
      }

      unmount()
    })
  })

  it('should handle empty search query', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} searchQuery="" />)

    const searchInput = screen.getByPlaceholderText('Search...')
    expect(searchInput).toHaveValue('')
  })

  it('should handle special characters in search query', () => {
    render(<ConfigurationSearchAndFilters {...defaultProps} searchQuery="test@#$%^&*()" />)

    const searchInput = screen.getByDisplayValue('test@#$%^&*()')
    expect(searchInput).toBeInTheDocument()
  })
})
