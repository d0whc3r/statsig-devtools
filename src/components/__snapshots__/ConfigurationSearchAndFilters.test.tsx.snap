// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfigurationSearchAndFilters > should match snapshot with default props 1`] = `
<div>
  <div
    class="flex-shrink-0 border-b border-gray-200 p-3"
  >
    <div
      class="mb-3"
    >
      <div
        class="relative"
      >
        <div
          class="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3"
        >
          <svg
            class="h-4 w-4 text-slate-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <input
          class="input-professional w-full py-2.5 pr-4 pl-10 text-sm"
          placeholder="Search..."
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <select
        class="input-professional min-w-0 flex-1 px-3 py-1.5 text-xs font-medium"
      >
        <option
          value="all"
        >
          All Types
        </option>
        <option
          value="feature_gate"
        >
          🚪 Gates
        </option>
        <option
          value="experiment"
        >
          🧪 Experiments
        </option>
        <option
          value="dynamic_config"
        >
          ⚙️ Configs
        </option>
      </select>
      <select
        class="input-professional min-w-0 flex-1 px-3 py-1.5 text-xs font-medium"
      >
        <option
          value="all"
        >
          All Status
        </option>
        <option
          value="passed"
        >
          ✅ Passed
        </option>
        <option
          value="failed"
        >
          ❌ Failed
        </option>
      </select>
    </div>
    <div
      class="mt-3 flex items-center justify-between"
    >
      <div
        class="flex items-center gap-2 text-xs text-slate-600"
      >
        <svg
          class="h-3.5 w-3.5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 5H7a2 2 0 00-2 2v10a2 2 0 002 2h8a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <span
          class="font-medium"
        >
          <span>
            10
             configurations
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
